src/candle_components/num_trades.rs

```diff
@@ -1,30 +1,30 @@
+use std::marker::PhantomData;
+
 use crate::{CandleComponent, CandleComponentUpdate, TakerTrade};
 
 /// This 'CandleComponent' keeps track of the number of trades
 #[derive(Debug, Default, Clone)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub struct NumTrades {
-    value: f64,
+pub struct NumTrades<T> {
+    value: u32,
+    phantom: PhantomData<T>,
```

Comment:
Why is the `PhantomData` required here? I think it can be safely removed. The `u32` type should be fine as I don't expect more trades to be in a single candle such that the `u32::MAX` would be exceeded.

Reply:
Ah, yeah, this has to do with needing the type to be generic in order for the derive macros to work properly.  I will remove  `PhantomData` and just make `value` generic over T, like with `TimeStamp<T>`.  I had `PhantomData` so I could control the type explicitly to be `u32` for value in the definition while still having the derive work but it's technically not needed.  This is a little bit of a smell but not sure how to get around it and still support `derive`.